yazi-plugin/src/external/highlighter.rs

```diff
@@ -84,23 +86,40 @@ impl Highlighter {
 				buf.push(b'\n');
 			}
 
+			let line = String::from_utf8_lossy(&buf).into_owned();
+
 			if i > skip {
-				after.push(String::from_utf8_lossy(&buf).into_owned());
+				after.push(line);
 			} else if !plain {
-				before.push(String::from_utf8_lossy(&buf).into_owned());
+				before.push(line);
 			}
 			buf.clear();
 		}
 
-		if skip > 0 && i < skip + limit {
-			return Err(PeekError::Exceed(i.saturating_sub(limit)));
+		if skip > 0 && i < skip + area.height as usize {
+			return Err(PeekError::Exceed(i.saturating_sub(area.height as usize)));
```

Comment:
Here we need to consider the number of lines after wrapping; otherwise, it will cause the [`seek` command](https://yazi-rs.github.io/docs/configuration/keymap/#manager.seek) abnormal, i.e. making it impossible to scroll to the very end of the file.

Reply:
Good catch! Unfortunately, after analyzing the issue I have to admit that it is way above my pay grade. The complexity introduced by preview scrolling makes the implementation much more difficult than my close to zero Rust skills supported by LLM can handle.

I'm sorry but I have to capitulate here and leave the baton for the more skilled developer to pick up.